### PR TITLE
ttygif init at 1.4.0

### PR DESCRIPTION
**Pull request by moaxcp**
###### Motivation for this change

add ttygif to nixpkgs

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/tools/misc/ttygif/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/icholy/ttygif/archive/1.4.0.tar.gz";
+    sha256 = "0apf7j6291rgv4s7c58cjq284ldc3scsws0003wdqayyr9fxr8vc";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ttygif $out/bin
+  '';
```
Comment (jonringer) [inline pkgs/tools/misc/ttygif/default.nix]: use the pre-existing Makefile, replace installPhase with
```suggestion
  makeFlags = [ "PREFIX=${placeholder "out"}" ];
```

File: pkgs/tools/misc/ttygif/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "ttygif";
+  version = "1.4.0";
+
+  src = fetchurl {
```
Comment (jonringer) [inline pkgs/tools/misc/ttygif/default.nix]: The fetchFromGitHub helper is usually prefered. It's especially helpful if we want to select commits off master.
```
  src = fetchFromGitHub {
    owner = "icholy";
    repo = pname;
    rev = version;
    sha256 = "0apf7j6291rgv4s7c58cjq284ldc3scsws0003wdqayyr9fxr8vc";
  };
```

File: pkgs/tools/misc/ttygif/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/tools/misc/ttygif/default.nix]: then the default `make install` command will work as intended.

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executable seems to work

```
[3 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/71007
1 package were build:
ttygif
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build ttygif

**Review by matthewbauer — CHANGES_REQUESTED** [review]
Please mark this as broken on x86_64-darwin.

**Comment by moaxcp** [comment]
@matthewbauer I not sure if I did this right. I set the platform to platforms.linux. Does this work or did you have something else in mind?

**Comment by jonringer** [comment]
that works

**Comment by jonringer** [comment]
@GrahamcOfBorg build ttygif

**Pull request merged**